lib/middleman-prismic/preview.rb

```diff
@@ -0,0 +1,22 @@
+module Middleman
+  module Prismic
+    class Preview
+      def initialize(app, options={})
+        @app = app
+      end
+
+      def call(env)
+        req = ::Rack::Request.new(env)
+        if req.path =~ /^\/preview/
+          token = req.params["token"]
+
+          Kernel.system "middleman prismic --ref #{token}"
```

Comment:
Does this override the whole content of the site with the preview content specified by the token? If so, what happens to other visitors to the site, will they also see the preview content?

Reply:
Yes, it does. So, the intention and requirement with Middleman w/ Prismic is that you run a separate "preview" site that is just for previewing content. Anyone that visits the preview site will see the same content as other visitors. All content will be replaced lived when the `/preview` route it hit.